### ISSUE TITLE
Malformed SwiftLint directives should generate an error or warning

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#4539]
### New Issue Checklist

- [ x] Updated SwiftLint to the latest version
- [x ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I've just been reviewing our code base for "blanket disables" where an engineer disabled some rule in a stretch of code (or at the top of the file) and then "forgot" to enable it after the relevant stretch of code.

I came across quite a few typos where someone (possibly even me) had written

`// swiftlint:disaable some_rule`

or more commonly

`// swiftlint:enaable some_rule`

which kind of makes sense, because an un-disabled rule that's needed would likely be noticed straight away, but an un-reenabled rule could remain silently re-enabled forever.

It would be great if SwiftLint could print a warning when it sees `// swiftlint:something` where `something` is not a valid directive.

##### Complete output when running SwiftLint, including the stack trace and command used

N/A

### Environment

N/A

**Comment by jpsim** [comment]
Great idea. This would be a great first issue if you felt inclined to contribute this to SwiftLint yourself.

**Comment by mildm8nnered** [comment]
So I think I can see where I can detect these, in `Command` `init`, so something like (very hackily)

```
    public init?(actionString: String, line: Int, character: Int) {
        let scanner = Scanner(string: actionString)
        _ = scanner.scanString("swiftlint:")
        // (enable|disable)(:previous|:this|:next)
        guard let actionAndModifierString = scanner.scanUpToString(" ") else {
            queuedPrintError("Found a swiftlint comment directive without an action")
            return nil
        }
        let actionAndModifierScanner = Scanner(string: actionAndModifierString)
        guard let actionString = actionAndModifierScanner.scanUpToString(":") else {
            queuedPrintError("Found a swiftlint comment directive without an action")
            return nil
        }        
        guard let action = Action(rawValue: actionString) else {
            queuedPrintError("Found a swiftlint comment with an invalid action")
            return nil
        }
```

And I have the line number to use in the message.

What I couldn't work out was how to get the filename, as without that the error message is not that much use ("somewhere in your files there is a malformed directive ...).

That initialiser seems to get called from here:

```
private extension Trivia {
    func commands(offset: AbsolutePosition, locationConverter: SourceLocationConverter) -> [Command] {
        var triviaOffset = SourceLength.zero
        var results: [Command] = []
        for trivia in self {
            triviaOffset += trivia.sourceLength
            switch trivia {
            case .lineComment(let comment), .blockComment(let comment):
                if
                    let lower = comment.range(of: "swiftlint:")?.lowerBound,
                    case let actionString = String(comment[lower...]),
                    case let end = locationConverter.location(for: offset + triviaOffset),
                    let line = end.line,
                    let column = end.column,
                    let command = Command(actionString: actionString, line: line, character: column)
                {
                    results.append(command)
                }
            default:
                break
            }
        }

        return results
    }
}
```

SourceLocationConverter has a file property, but it is not public :-(

**Comment by mildm8nnered** [comment]
So it looks like `end` (from `case let end = locationConverter.location(for: offset + triviaOffset)`) has a `file` property ...
`

**Comment by mildm8nnered** [comment]
https://github.com/realm/SwiftLint/pull/4543

**Comment by mildm8nnered** [comment]
OR https://github.com/realm/SwiftLint/pull/4546, which implements bad SwiftLint command syntax as a rule.

**Comment by mildm8nnered** [comment]
Resolved by #4546
